includes/commerce_pos.transaction.inc

```diff
@@ -197,6 +197,64 @@ function commerce_pos_transaction_form(&$form, &$form_state, $transaction_type)
 
       $line_item_element = &$form['line_items'][$line_item_id];
 
+      // Determine if the product has sufficient stock if it was enabled for
+      // this product.
+      $no_stock = FALSE;
+      if (isset($line_item_wrapper->commerce_product->value()->commerce_stock) &&
```

Comment:
There is probably more abstract ways to check for stock that doesn't rely on a field. At the very least we should make our checks here alterable with a hook_alter() call so people implementing this can help determine what the stock value truly is.